feat: add instance creator and handle velocity event

Jarva opened the pull request:
Adds the ability to create instances of PlaceholderAPI to manage caching separately

Also fixes an issue where the Velocity event for the placeholder was still forwarded to the client

WiIIiam278 commented on common/src/main/java/net/william278/papiproxybridge/api/PlaceholderAPI.java:
```diff
@@ -90,7 +87,16 @@ public static PlaceholderAPI getInstance() {
+    }
+
+    /**
+     * Create a new instance of PlaceholderAPI allowing unique customisation of caching mechanisms
+     * @return PlaceholderAPI instance that can be used to format text
```
jd style nit: space before `@returns` line.
Perhaps also an `@since 1.3`?

Jarva replied:
Resolved

WiIIiam278 commented on common/src/main/java/net/william278/papiproxybridge/api/PlaceholderAPI.java:
```diff
@@ -90,7 +87,16 @@ public static PlaceholderAPI getInstance() {
+     * Create a new instance of PlaceholderAPI allowing unique customisation of caching mechanisms
+     * @return PlaceholderAPI instance that can be used to format text
+     */
+    public static PlaceholderAPI createInstance() {
+        return new PlaceholderAPI();
     }
```
could this method be moved above `register`?

Jarva replied:
Moved

WiIIiam278 reviewed (changes requested):
I'd say this is fairly needed so happy to go with this :+1:. Some comments, though.

It's a bit weird that we're supporting both the singleton and multi-instance-based approach, but I suppose the alternative would be deprecating `getInstance()`. 

Might be worth updating the javadocs for the `getInstance()` and above the class though to explain this. Perhaps also recommend the new method in the README?

That aside, great work.

Jarva commented:
I've updated the documentation to mention the new method in getInstance and updated the readme to reflect the multi-instance based approach, I'd prefer to deprecate getInstance, alternatively we could transparently change getInstance to provide a new instance each time? Shouldn't impact usage given only one plugin uses expiry currently?

WiIIiam278 commented:
> alternatively we could transparently change getInstance to provide a new instance each time? Shouldn't impact usage given only one plugin uses expiry currently?

This would break setups that call `#getInstance()` each time to set the cache time / make a placeholder query rather than saving the instance. 

> given only one plugin uses expiry currently?

Not sure exactly how popular this feature is, but it is a relatively smaller feature at least. Tbh though, I suspect devs calling the `setCacheExpiry` method probably aren't running getInstance() _every_ time (at least, I'd like to think); it also wouldn't be particularly breaking as correcting the functionality.

I suggest we deprecate `getInstance()`, add an `@apiNote` about this change to the JavaDocs & make it clear in the changelogs, and have `getInstance()` start serving unique instances (mirroring `createInstance()`)

Jarva commented:
Updated as requested
